### PR TITLE
Rename SlashCommandInteractionOptionsProvider methods

### PR DESCRIPTION
**Pull request by KILLEliteMaste**
## Checklist
- [x] I have tested this PR[^1].
- [x] I have read the [contributing guidelines](./CONTRIBUTING.md).

### Breaking Changes
- Renamed SlashCommandOptionsProvider methods that are only usable for arguments of a slash command from `getOption*` to `getArgument*` and changed the implementation from getting the options from the option to getting always the command arguments if there are any

## Description
See issue for more detailed thoughts on this

Closes: #1116 

[^1]: At least started a running bot instance with your changes and triggered an event so your changed code gets executed.

### CODE REVIEW

File: javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteraction.java [hunk 1]
```diff
@@ -1,19 +1,7 @@
...
 
-    /**
-     * Gets the arguments of this slash command if there are any.
-     *
-     * <p>This is a shorthand method to avoid checking for Subcommmands or SubcommandGroups
-     * to get the slash command arguments.
-     *
-     * @return The argument options.
-     */
-    List<SlashCommandInteractionOption> getArguments();
```
Comment (Saladoc) [inline javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteraction.java:15]: Is there no way to keep / deprecate this method for another release?

File: build.gradle.kts [hunk 2]
```diff
@@ -6,8 +6,8 @@ plugins {
     `java-library`
     `maven-publish`
     signing
-    id("com.github.johnrengelman.shadow") version "2.0.4" apply false
-    id("net.researchgate.release") version "2.7.0" apply false
+    id("com.github.johnrengelman.shadow") version "7.1.2" apply false
+    id("net.researchgate.release") version "3.0.2" apply false
```
Comment (Saladoc) [inline build.gradle.kts:10]: Sneay version upgrades not mentioned in any commit or pr description - should probably be a separate commit?

File: javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOptionsProvider.java [hunk 3]
```diff
@@ -37,129 +47,142 @@ default Optional<SlashCommandInteractionOption> getOptionByName(String name) {
...
     }
 
     /**
-     * Gets the string value of an option having the specified name.
+     * Gets the string representation value of an argument having the specified name.
      *
-     * @param name The name of the option to search for.
-     * @return An Optional with the string value of such an option if it exists; an empty Optional otherwise
+     * @param name The name of the argument to search for.
+     * @return An Optional with the string value of such an argument if it exists; an empty Optional otherwise
```
Comment (Saladoc) [inline javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOptionsProvider.java:66]: Can we get the difference between a string representation and a string value mentioned in the javadoc? The `@return` tag is the same as for `#getStringValueByName(String)` which might be confusing.

**Review by Saladoc — CHANGES_REQUESTED** [review]

File: javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteraction.java (hunk 1 above)
Comment (KILLEliteMaste) [reply javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteraction.java:15]: There shouldn't be any noticeable difference for users except when using reflection as it implements the `SlashCommandINteractionOptionsProvider`

File: build.gradle.kts (hunk 2 above)
Comment (KILLEliteMaste) [reply build.gradle.kts:10]: Yeah might be better :) Will do it later if it gets approved

File: javacord-core/build.gradle.kts [hunk 4]
```diff
@@ -2,18 +2,21 @@ import aQute.bnd.version.MavenVersion.parseMavenString
...
     mavenCentral()
+    maven(url="https://m2.dv8tion.net/releases")
 }
 
 dependencies {
     api(project(":javacord-api")) {
         isTransitive = true
     }
 
+    implementation("com.sedmelluq:lavaplayer:1.3.77")
```
Comment (Saladoc) [inline javacord-core/build.gradle.kts:18]: Where is this dependency used? Is this a leftover from testing another branch?

**Review by Saladoc — APPROVED** [review]
